features/api-layer/src/main/java/org/opennms/features/apilayer/config/ThreshdConfigurationExtensionManager.java

```diff
@@ -0,0 +1,156 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.apilayer.config;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.opennms.integration.api.v1.config.thresholding.PackageDefinition;
+import org.opennms.integration.api.v1.config.thresholding.ThreshdConfigurationExtension;
+import org.opennms.netmgt.config.dao.thresholding.api.WriteableThreshdDao;
+import org.opennms.netmgt.config.threshd.ExcludeRange;
+import org.opennms.netmgt.config.threshd.Filter;
+import org.opennms.netmgt.config.threshd.IncludeRange;
+import org.opennms.netmgt.config.threshd.Package;
+import org.opennms.netmgt.config.threshd.Parameter;
+import org.opennms.netmgt.config.threshd.Service;
+import org.opennms.netmgt.config.threshd.ServiceStatus;
+import org.opennms.netmgt.config.threshd.ThreshdConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ThreshdConfigurationExtensionManager extends ConfigExtensionManager<ThreshdConfigurationExtension,
+        ThreshdConfiguration> {
+    private static final Logger LOG = LoggerFactory.getLogger(ThreshdConfigurationExtensionManager.class);
+
+    private final WriteableThreshdDao threshdDao;
+
+    public ThreshdConfigurationExtensionManager(WriteableThreshdDao threshdDao) {
+        super(ThreshdConfiguration.class, new ThreshdConfiguration());
+        this.threshdDao = Objects.requireNonNull(threshdDao);
+        LOG.debug("ThreshdConfigurationExtensionManager initialized.");
+    }
+
+    @Override
+    protected ThreshdConfiguration getConfigForExtensions(Set<ThreshdConfigurationExtension> extensions) {
+        ThreshdConfiguration mergedConfig = new ThreshdConfiguration();
+
+        mergedConfig.setPackages(extensions.stream()
+                .flatMap(e -> e.getPackages().stream())
+                .map(ThreshdConfigurationExtensionManager::toPackage)
+                .collect(Collectors.toList()));
+
+        return mergedConfig;
+    }
+
+    @Override
+    protected void triggerReload() {
+        LOG.debug("Threshd configuration changed. Triggering a reload.");
+        threshdDao.onConfigChanged();
```

Comment:
We also need to call `reinitializeThresholdingSets` on the `thresholdingSetPersister` used by the `ThresholdingServiceImpl` in order for the new/updated thresholds to be applied to existing sessions. See: https://github.com/OpenNMS/opennms/blob/5e22a62924f60adbac58cabe04c284e5790a0d55/features/collection/thresholding/impl/src/main/java/org/opennms/netmgt/threshd/ThresholdingServiceImpl.java#L240